homeassistant/components/snapcast/media_player.py

```diff
@@ -259,9 +260,15 @@ def source_list(self):
     @property
     def state(self):
         """Return the state of the player."""
-        if self._client.connected:
-            return STATE_ON
-        return STATE_OFF
+        if not self._client.connected:
+            return STATE_OFF
+        if self.is_volume_muted or self._client.group.muted:
+            return STATE_IDLE
```

Comment:
Snapcast clients or groups do not have something like an on/off or activated/deactivated state. Also there is always one stream selected. So the only way to prevent a client from playing is to mute it. Here is an example: There is only one stream defined in the snapcast server (e.g. from mpd player) and there are two snapcasts clients (A and B). So both, A and B, are connected to the same stream. If you want to play music on A you need to start the stream and unmute A. To prevent B from playing the same stream, B needs to be muted. The (un)muted state is controlled by the server, so the server will not send any data to a muted client.